core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java

```diff
@@ -81,7 +81,8 @@ public void blockHeartbeat(BlockHeartbeatPRequest request,
             .stream()
             .collect(
                 Collectors.toMap(
-                    e -> Block.BlockLocation.newBuilder().setTier(e.getKey().getTierAlias())
+                    e -> Block.BlockLocation.newBuilder().setWorkerId(workerId)
+                        .setTier(e.getKey().getTierAlias())
```

Comment:
We have the worker ID here already, so we can just set it to the `BlockLocation` and pass it to the `processWorkerAddedBlocks()` method. The worst case is, the blocks are unknown and we just ignore them.

I don't really see a reason why we ignore the worker ID here. The worker IDs don't change anywhere in the `register()` or `heartbeat()`. We just look for the `MasterWorkerInfo` with the ID and then get the ID from the `MasterWorkerInfo` again... Another reason might be we don't add the worker ID here so the hash code doesn't contain worker ID, so the `BlockLocation` items are together with other `BlockLocation` from other workers in some map. But that also doesn't make sense...